ffi/diplomat/src/collator.rs

```diff
@@ -0,0 +1,238 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+#[diplomat::bridge]
+pub mod ffi {
+    use alloc::boxed::Box;
+    use diplomat_runtime::DiplomatResult;
+    use icu_collator::{Collator, CollatorOptions};
+
+    use crate::{
+        errors::ffi::ICU4XError, locale::ffi::ICU4XLocale, provider::ffi::ICU4XDataProvider,
+    };
+
+    #[diplomat::opaque]
+    #[diplomat::rust_link(icu_collator::Collator, Struct)]
+    pub struct ICU4XCollator(pub Collator);
+
+    #[diplomat::enum_convert(core::cmp::Ordering)]
+    #[diplomat::rust_link(core::cmp::Ordering, Enum)]
+    pub enum ICU4XOrdering {
+        Less = 0,
+        Equal = 1,
+        Greater = 2,
+    }
```

Comment:
We can't change the repr over Diplomat just yet.

But it can still have negative values just fine, without a repr.

Reply:
I tried changing the discriminants from `0, 1, 2` to `-1, 0, 1`, and I got this error when running `cargo make diplomat-gen`:
```
thread 'main' panicked at 'Expected a discriminant to be a constant integer', core/src/ast/enums.rs:45:33
```
When I revert back to `0, 1, 2`, then `diplomat-gen` completes successfully.

Reply:
Ah. I think that's an artificial restriction in Diplomat that ought to be fixed.

Reply:
My total newb blind guess diagnosis is that it is because `-1` has to be treated as an [`ExpUnary`](https://docs.rs/syn/latest/syn/struct.ExprUnary.html) plus a [`LitInt`](https://docs.rs/syn/latest/syn/struct.LitInt.html), which is why just the `LintInt` alone is failing to parse.

Reply:
https://github.com/rust-diplomat/diplomat/pull/255

Reply:
That merged

Perhaps we can fix this in a followup so that you don't have to deal with a diplomat upgrade>

Reply:
We did the diplomat upgrade.